### PR TITLE
Adding reconfiguration support for RO Replica

### PR DESCRIPTION
**Pull request by nkumar04**
In this PR we add the following changes
- we add changes to addRemove reconfiguration command with epoch number in reserved pages. 
- Store latest reconfiguration command block in reserved pages (for ROR reconfiguration)
- Added api to read and store last stored reconfiguration blockid from RO meatadata
- Added new ro reconfiguration state client and handler for cre in ro replica
- Load reconfiguration command from reserved pages and execute it

### CODE REVIEW

File: bftengine/include/bftengine/ReconfigurationCmd.hpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+// file.
+
+#pragma once
+#include "ReservedPagesClient.hpp"
+#include "Serializable.h"
+#include "concord.cmf.hpp"
+#include <optional>
+
+namespace bftEngine {
+class ReconfigurationCmd : public bftEngine::ResPagesClient<ReconfigurationCmd, 1> {
```
Comment (yontyon) [inline bftengine/include/bftengine/ReconfigurationCmd.hpp:21]: What if we have several reconfiguration requests that are relevant to this epoch, wouldn't it be better to save a list of the actual reconfiguration requests (and our current epoch number)

**Review by yontyon — APPROVED** [review]
LGTM!
I only have one suggestion for a change
thanks

File: bftengine/include/bftengine/ReconfigurationCmd.hpp (hunk 1 above)
Comment (nkumar04) [reply bftengine/include/bftengine/ReconfigurationCmd.hpp:23]: fixed

**Review by yontyon — APPROVED** [review]
LGTM!

File: client/reconfiguration/CMakeLists.txt [hunk 2]
```diff
@@ -3,9 +3,12 @@ project (cre LANGUAGES CXX)
 add_library(cre
         src/client_reconfiguration_engine.cpp
         src/poll_based_state_client.cpp
+        src/ror_reconfiguration_client.cpp
```
[the remark this answers is not in the document]
Comment (nkumar04) [reply client/reconfiguration/CMakeLists.txt]: fixed

File: client/reconfiguration/CMakeLists.txt [hunk 3]
```diff
@@ -3,9 +3,12 @@ project (cre LANGUAGES CXX)
...
+        src/ror_reconfiguration_handler.cpp
         )
 
-target_include_directories(cre PUBLIC include)
+target_include_directories(cre PUBLIC include
```
[the remark this answers is not in the document]
Comment (nkumar04) [reply client/reconfiguration/CMakeLists.txt]: fixed

**Pull request merged**